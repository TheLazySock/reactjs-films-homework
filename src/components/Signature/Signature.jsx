import React from 'react';
import PropTypes from 'prop-types';
import style from './style.scss';

const Signature = props => (
  <div className={ style.background }>
    <h3 className={ style.name }>
      {props.name}
    </h3>
  </div>
);

Signature.propTypes = {
  name: PropTypes.string.isRequired,
};

export default Signature;
