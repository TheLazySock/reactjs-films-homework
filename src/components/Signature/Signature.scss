.background {
  position: relative;
  width: 100vw;
  height: 100vh;
  background: #dfb3bc;
  animation: grad 8s linear infinite alternate;
}

@keyframes grad {
  50% {
    background: #a9e1e2;
  }
}

.name {
  color: #d83939;
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  font-size: 60px;
}
