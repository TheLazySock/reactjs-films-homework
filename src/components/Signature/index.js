export { default } from './Signature.jsx';
