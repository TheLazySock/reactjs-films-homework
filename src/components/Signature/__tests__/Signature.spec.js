import React from 'react';
import renderer from 'react-test-renderer';

import Signature from '../index';


describe('Signature', () => {
  it('rendered with correct name', () => {
    const output = renderer.create(<Signature name="Frodo Baggins" />);
    expect(output).toMatchSnapshot();
  });
});
