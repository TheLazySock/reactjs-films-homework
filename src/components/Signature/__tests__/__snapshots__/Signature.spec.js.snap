// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signature rendered with correct name 1`] = `
<div
  className="background"
>
  <h3
    className="name"
  >
    Frodo Baggins
  </h3>
</div>
`;
